configs/xgb_gpu_config.json

```diff
@@ -1,160 +1,203 @@
                 }
             ],
-            "n-estimators": [60],
-            "objective": ["multi:softprob"],
-            "tree-method": ["gpu_hist"],
-            "max-depth": [7],
-            "subsample": [0.7],
-            "colsample-bytree": [0.7]
+            "learning-rate":    0.03,
+            "max-depth":        6,
+            "n-estimators":     1000,
+            "objective":        "reg:squarederror"
         },
         {
-            "algorithm": "gbt",
             "dataset": [
                 {
-                    "source": "csv",
-                    "name": "santander",
+                    "source":   "npy",
+                    "name":     "airline-ohe",
                     "training":
                     {
-                        "x": "data/santander_x_train.csv",
-                        "y": "data/santander_y_train.csv"
+                        "x":    "data/airline-ohe_x_train.npy",
+                        "y":    "data/airline-ohe_y_train.npy"
+                    },
+                    "testing":
+                    {
+                        "x":    "data/airline-ohe_x_test.npy",
+                        "y":    "data/airline-ohe_y_test.npy"
+                    }
+                }
+            ],
+            "reg-alpha":        0.9,
+            "max-bin":          256,
+            "scale-pos-weight": 2,
+            "learning-rate":    0.1,
+            "subsample":        1,
+            "reg-lambda":       1,
+            "min-child-weight": 0,
+            "max-depth":        8,
+            "max-leaves":       256,
+            "n-estimators":     1000,
+            "objective":        "binary:logistic"
+        },
+        {
+            "dataset": [
+                {
+                    "source":   "npy",
+                    "name":     "higgs1m",
+                    "training":
+                    {
+                        "x":    "data/higgs1m_x_train.npy",
+                        "y":    "data/higgs1m_y_train.npy"
+                    },
+                    "testing":
+                    {
+                        "x":    "data/higgs1m_x_test.npy",
+                        "y":    "data/higgs1m_y_test.npy"
                     }
                 }
             ],
-            "n-estimators": [10000],
-            "objective": ["binary:logistic"],
-            "tree-method": ["gpu_hist"],
-            "max-depth": [1],
-            "subsample": [0.5],
-            "eta": [0.1],
-            "colsample-bytree": [0.05]
+            "reg-alpha":        0.9,
+            "max-bin":          256,
+            "scale-pos-weight": 2,
+            "learning-rate":    0.1,
+            "subsample":        1,
+            "reg-lambda":       1,
+            "min-child-weight": 0,
+            "max-depth":        8,
+            "max-leaves":       256,
+            "n-estimators":     1000,
+            "objective":        "binary:logistic",
+            "inplace-predict":  ""
```

Comment:
It will be good to replace valueless flags with some special fake value in json like `"inplace-predict": "SET_FLAG_WITHOUT_VALUE"` for better readability

Reply:
It's definitely good idea, but I think we don't have enough time to do so as part of this task. I don't see a fast way to implement this now.
And there are lots of other ideas waiting to be implemented, as we discussed with @PetrovKP. Let's find a time and do it, I'm ready to join